src/quacc/runners/ase.py

```diff
@@ -199,6 +199,8 @@ def run_opt(
             },
             optimizer_kwargs,
         )
+        if optimizer_kwargs is None:
+            optimizer_kwargs = {}
```

Comment:
The issue stems from `optimizer_kwargs` being typed as `dict[str, None] | None`. Despite `recursive_dict_merge` assigning a dictionary to `optimizer_kwargs`, mypy still considers it potentially `None` because of its initial typing declaration. That's why I added the check

Reply:
Interesting. I wonder why that is. Is it because we are overwriting `optimizer_kwargs` with a variable of the same name? If so, perhaps we should be using a new name instead? It seems odd that if `recursive_dict_merge` doesn't return `None` in its type hint that mypy would think `optimizer_kwargs` could be `None`.

Reply:
Yes, using a different name should resolve the issue. Another potential solution is to use an assert statement. Since we are confident that optimizer_kwargs will not be None, we can include the following in the code:
```python
assert optimizer_kwargs is not None
```
what do you think ?

Reply:
Personally, I think a different name would be wiser. Perhaps `merged_optimizer_kwargs`? This might be less prone to error than something like `optimizer_kwargs_`.

Having an `assert` statement in the code should really only be there as an internal way to say "this _might_ conceivably happen, but it really shouldn't." In reality, it is actually impossible for it to be `None` based on the way the code is written. There is no scenario where it would be `None` here.